cartoframes/assets/src/map/popups.js

```diff
@@ -51,9 +51,12 @@ export function updatePopup(map, popup, event, attrs) {
         .setLngLat([event.coordinates.lng, event.coordinates.lat])
         .setHTML(`<div class="popup-content">${popupHTML}</div>`);
 
-    if (popupHTML.length > 0 && !popup.isOpen()) {
+    if (!popup.isOpen()) {
```

Comment:
I would avoid setting and adding, and removing the popup every time. We could try something like this instead:

```py
if (popupHTML) {
    popup
        .setLngLat([event.coordinates.lng, event.coordinates.lat])
        .setHTML(`<div class="popup-content">${popupHTML}</div>`);

    if (!popup.isOpen()) {
      popup.addTo(map);
    }
} else {
    popup.remove();
}
```